cmp/form/NumberField.js

```diff
@@ -24,22 +25,37 @@ export class NumberField extends HoistField {
         /** minimum value */
         min: PT.number,
         /** maximum value */
-        max: PT.number
+        max: PT.number,
+        /** Number of decimal places to allow on field's value, defaults to 4 */
+        precision: PT.number,
+        /** Allow/automatically fill in trailing zeros in accord with precision, defaults to false */
+        zeroPad: PT.bool,
+        /** Set to true for advanced input evaluation, defaults to false.
+            Inputs suffixed with k, m, or b will be calculated as thousands, millions, or billions respectively */
+        supportShorthandUnits: PT.bool,
+        /** Whether to display large values with commas */
+        displayWithDelimiters: PT.bool,
```

Comment:
Kinda wonky name on this one, especially as it might be interpreted as affecting the decimal point or some other kind of delimiter.  Would `displayWithCommas` or `useCommas` or `addCommas` be better?